src/explorer/CosmosDBConnection.ts

```diff
@@ -47,15 +47,16 @@ export class CosmosDBConnection extends AzureTreeItem<ISiteTreeRoot> {
     }
 
     public async deleteTreeItemImpl(): Promise<void> {
-        const appSettings = await this.root.client.listApplicationSettings();
+        const appSettingsClient: IAppSettingsClient = this.parent.parent.client;
```

Comment:
Should probably make `CosmosDBConnection` extend `AzExtTreeItem` so that we can be sure it doesn't try to use `root` anywhere else